### PR TITLE
XFAIL: lld-available lit.cfg tag does not work as intended.

### PR DESCRIPTION
**Pull request by kamaub**
This change XFAILs instrprof-gc-sections.c compiler-rt test case
because it uses the `lld-available` lit.config tag which erroneously
uses the `CMAKE_LINKER` if it is lld instead of checking for the
latest and greatest lld available. A patch is in the works to fix
this.
This change is to bring the clang-ppc64le-linux-test-suite and
clang-ppc64le-linux-multistage back to green as they were broken
as of PR #98382

### CODE REVIEW

File: compiler-rt/test/profile/instrprof-gc-sections.c [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // REQUIRES: linux, lld-available
 
 // FIXME: Investigate and fix.
-// XFAIL: powerpc64-target-arch
+// XFAIL: powerpc64-target-arch,powerpc64le-target-arch
```
Comment (lei137) [inline compiler-rt/test/profile/instrprof-gc-sections.c:4]: This is only an issue on the 64bit arch? What about 32bit?

Comment (kamaub) [reply compiler-rt/test/profile/instrprof-gc-sections.c:4]: It affects anyone that had lld in their environment but but does not want to test it, this is only fixes the effected bots.

Comment (kamaub) [inline compiler-rt/test/profile/instrprof-gc-sections.c:4]:
```suggestion
// XFAIL: powerpc64-target-arch, powerpc64le-target-arch
```

**Review by lei137 — APPROVED** [review]
thx

**Review by amy-kwan — APPROVED** [review]

**Pull request merged**